src/stream/stream/mod.rs

```diff
@@ -797,27 +816,28 @@ extension_trait! {
             #
             use async_std::prelude::*;
             use std::collections::VecDeque;
-            use std::sync::mpsc::channel;
 
-            let (tx, rx) = channel();
-
-            let s: VecDeque<usize> = vec![1, 2, 3].into_iter().collect();
-            let sum = s.for_each(move |x| tx.clone().send(x).unwrap()).await;
+            let mut x = 0;
 
-            let v: Vec<_> = rx.iter().collect();
+            let s: VecDeque<_> = vec![1u8, 2, 3].into_iter().collect();
+            s.for_each(|item| {
+                x += item;
+                futures::future::ready(())
+            }).await;
```

Comment:
Could we perhaps use an async block here?

Reply:
capturing in async closures doesn't look that well now, but I will cook an example with Arc and Mutex

Reply:
@montekki oh dang, yeah you're right -- I think what we had before is def better indeed. Perhaps we should revert it?

This is def yet another argument for having async closures!

Reply:
done